pkg/async/pool/pool.go

```diff
@@ -227,9 +227,9 @@ func (p *Pool) worker(ctx context.Context) {
 
 // Close blocks until all workers finshes current items and terminates
 func (p *Pool) Close() {
-	p.wg.Wait()
 	p.cancel(&orerr.ShutdownError{Err: context.Canceled})
```

Comment:
@pavelsmejkal canceling the pool before the wait group completes was the cause of the original race condition in this library. If this is the order that we need to do things in, we should probably use a mutex to ensure that `ctx.Err` doesn't cause a race condition.

Reply:
@artemchernyak-outreach, true i was just leaving the office and didn't wait for all tests. But then it is not a problem of the pool but of the CancelWithError. Submitted the fix